Assets/LeapMotionModules/UI/Interaction/ActivityManager.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Leap.Unity.Query;
+using UnityEngine;
+
+namespace Leap.Unity.UI.Interaction {
+
+  /// <summary> The Activity Manager encapsulates the logic that determines which
+  /// InteractionBehaviours are within an interactible distance and which interactions
+  /// they should be considered for (hovering/contact/grasping).
+  /// 
+  /// Essentially, the ActivityManager is a wrapper around PhysX sphere queries around
+  /// Hands for InteractionBehaviours.</summary>
+  public class ActivityManager {
+
+    public float activationRadius;
+    public InteractionManager manager;
+
+    private Collider[] _colliderResultsBuffer = new Collider[32];
+    private HashSet<InteractionBehaviourBase> _activeBehaviours = new HashSet<InteractionBehaviourBase>();
+
+    public HashSet<InteractionBehaviourBase> ActiveBehaviours {
+      get { return _activeBehaviours; }
+    }
+
+    public ActivityManager(InteractionManager manager, float activationRadius) {
+      this.manager = manager;
+      this.activationRadius = activationRadius;
+    }
+
+    public void FixedUpdateHand(Hand hand) {
+      int count = GetSphereColliderResults(hand, _colliderResultsBuffer, out _colliderResultsBuffer);
+      UpdateActiveList(count, _colliderResultsBuffer);
+    }
+
+    private int GetSphereColliderResults(Hand hand, Collider[] resultsBuffer_in, out Collider[] resultsBuffer_out) {
+      resultsBuffer_out = resultsBuffer_in;
+      if (hand == null) return 0;
+
+      int overlapCount = 0;
+      while (true) {
+        overlapCount = Physics.OverlapSphereNonAlloc(hand.PalmPosition.ToVector3(),
+                                                         activationRadius * 100,
+                                                         resultsBuffer_in,
+                                                         ~0,
```

Comment:
Isn't this able to constrain the search based on the layer specified in the manager?